### ISSUE TITLE
NSFW detected ?

### ISSUE THREAD

**Issue by Username69992872** [s0md3v/sd-webui-roop#4]
Seriosly in Automatic1111 you can not add a NSFW detector. People want to work in freedom. I tested with some normal images, there are plenty of false detections.

**Comment by ArchAngelAries** [comment]
Agreed. This filter is more sensitive than Midjourney's. I'd like to know how to disable it because I want to make images to train on my AI Nobody personality but this filter trips on anything that's not 100% covered from the neck down.

**Comment by Disorbs** [comment]
it could be roop ik it has a detection as well if anyone close to a bikini it wont generate. not sure about stablediffusion

**Comment by Username69992872** [comment]
extensions/sd-webui-roop/scripts/cimage.py:

`# def check_batch(x: torch.Tensor) -> torch.Tensor:
    # x_samples_ddim_numpy = x.cpu().permute(0, 2, 3, 1).numpy()
    # x_checked_image, _ = check_image(x_samples_ddim_numpy)
    # x = torch.from_numpy(x_checked_image).permute(0, 3, 1, 2)
    # return x

def check_batch(x):
    return x

# def convert_to_sd(img: Image) -> Image:
    # _, hs = check_image(np.array(img))
    # if any(hs):
        # img = (
            # img.resize((int(img.width * 0.1), int(img.height * 0.1)))
            # .resize(img.size, Image.BOX)
            # .filter(ImageFilter.BLUR)
        # )
    # return img


def convert_to_sd(img: Image):
    return img`

there you go :)

**Comment by Username69992872** [comment]
beside the NSFW thingy - this Addon is really amazing, great results, awesome work.
